packages/salesforcedx-apex-replay-debugger/src/core/handles.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+export class Handles<T> {
```

Comment:
This was copied from vscode-debugadapter. I needed to bring this over to add method that can copy by value. Handles is used to store stuff about frames and variables.